flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SourceOperatorStreamTask.java

```diff
@@ -116,6 +116,11 @@ public void init() throws Exception {
             } else {
                 return super.triggerCheckpointAsync(checkpointMetaData, checkpointOptions);
             }
+        } else if (checkpointOptions.getCheckpointType() == CheckpointType.FULL_CHECKPOINT) {
+            // see FLINK-25256
+            throw new IllegalStateException(
+                    "NO_CLAIM mode is not supported in combination with externally induced"
+                            + " sources. Please use either CLAIM or LEGACY mode.");
```

Comment:
I just notice that we implicitly imply here that FULL_CHECKPOINT and NO_CLAIM are the same things but that is not true since I suppose FULL_CHECKPOINT can be used for different purposes in the future.
Right now, I don't have any good solution for this since I can not find a different good place for checking this condition but maybe you have an idea or we can leave it as is since this message looks helpful for user.

Reply:
I doubt it is ever used for other purposes, tbh.

Reply:
I think FULL_CHECKPOINT might be useful if we hope to enable to trigger manual checkpoint in the future. It's not better to bound them together.

Reply:
I don't mind changing the exception message. I've updated it.